wagtail/admin/views/workflows.py

```diff
@@ -452,3 +472,146 @@ def enable_task(request, pk):
         return redirect(redirect_to)
     else:
         return redirect('wagtailadmin_workflows:edit_task', task.id)
+
+
+def get_chooser_context():
+    """construct context variables needed by the chooser JS"""
+    return {
+        'step': 'chooser',
+        'error_label': _("Server Error"),
+        'error_message': _("Report this error to your webmaster with the following information:"),
+        'tag_autocomplete_url': reverse('wagtailadmin_tag_autocomplete'),
+    }
+
+
+def get_task_result_data(task):
+    """
+    helper function: given a task, return the json data to pass back to the
+    chooser panel
+    """
+
+    return {
+        'id': task.id,
+        'name': task.name,
+        'edit_url': reverse('wagtailadmin_workflows:edit_task', args=[task.id]),
+    }
+
+
+def task_chooser(request):
+    task_models = get_task_types()
+    create_model = None
+    can_create = False
+
+    if task_permission_policy.user_has_permission(request.user, 'add'):
+        can_create = len(task_models) != 0
+
+        if len(task_models) == 1:
+            create_model = task_models[0]
+
+        elif 'create_model' in request.GET:
+            create_model = resolve_model_string(request.GET['create_model'])
+
+            if create_model not in task_models:
+                raise Http404
+
+    # Build task types list for "select task type" view
+    task_types = [
+        (model.get_verbose_name(), model._meta.app_label, model._meta.model_name)
+        for model in task_models
+    ]
+    # sort by lower-cased version of verbose name
+    task_types.sort(key=lambda task_type: task_type[0].lower())
+
+    # Build task type choices for filter on "existing task" tab
+    task_type_choices = [
+        (model, model.get_verbose_name())
+        for model in task_models
+    ]
+    task_type_choices.sort(key=lambda task_type: task_type[1].lower())
+
+    if create_model:
+        edit_handler = create_model.get_edit_handler()
+        edit_handler = edit_handler.bind_to(request=request)
+        createform_class = edit_handler.get_form_class()
+    else:
+        createform_class = None
+
+    # TODO
+    #  # allow hooks to modify the queryset
+    # for hook in hooks.get_hooks('construct_document_chooser_queryset'):
+    #     documents = hook(documents, request)
```

Comment:
Looks like a remnant from copying the document_chooser method. Would be good to cleanup